Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Domain/Action/UI/Message.hs

```diff
@@ -112,6 +116,16 @@ messageSeen :: Id SP.Person -> Id Domain.Message -> Flow APISuccess
 messageSeen driverId messageId = do
   _ <- Esq.runInReplica $ QP.findById driverId >>= fromMaybeM (PersonNotFound driverId.getId)
   Esq.runTransaction $ MRQ.updateSeenAndReplyByMessageIdAndDriverId messageId (cast driverId) True Nothing
+
+  return Success
+
+messageLiked :: Id SP.Person -> Id Domain.Message -> Flow APISuccess
+messageLiked driverId messageId = do
+  _ <- Esq.runInReplica $ QP.findById driverId >>= fromMaybeM (PersonNotFound driverId.getId)
+  Esq.runTransaction $ MRQ.updateMessageLikeByMessageIdAndDriverIdAndReadStatus messageId (cast driverId)
+  messageDetails <- Esq.runInReplica $ MRQ.findByMessageIdAndDriverId messageId (cast driverId) >>= fromMaybeM (InvalidRequest "Message not found")
```

Comment:
You shouldn't use read query from replica db right after transaction, because value can update very slowly.

Instead you can fetch all required values from replica DB, and then update all required values within one transaction

Reply:
ok